include/deal.II/particles/particle_accessor.h

```diff
@@ -445,10 +506,13 @@ namespace Particles
   ParticleAccessor<dim, spacedim>::get_surrounding_cell(
     const Triangulation<dim, spacedim> &triangulation) const
   {
-    Assert(particle != map->end(), ExcInternalError());
+    Assert(active_cell_index < particles->size() &&
+             particle_index < (*particles)[active_cell_index].size(),
+           ExcInternalError());
 
-    const typename Triangulation<dim, spacedim>::cell_iterator cell(
-      &triangulation, particle->first.first, particle->first.second);
+    typename Triangulation<dim, spacedim>::active_cell_iterator cell =
+      triangulation.begin_active();
+    std::advance(cell, active_cell_index);
```

Comment:
I guess we should cache in `Triangulation` a translation between active_cell_index and a level/index pair.